src/main/java/com/hivemq/cli/commandline/CommonErrorMessageHandler.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2019 HiveMQ and the HiveMQ Community
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.hivemq.cli.commandline;
+
+import org.jetbrains.annotations.NotNull;
+import picocli.CommandLine;
+
+import java.io.PrintWriter;
+
+abstract class CommonErrorMessageHandler implements CommandLine.IParameterExceptionHandler {
+
+    CommandLine cmd;
+    PrintWriter writer;
+    private CommandLine.ParameterException parameterException;
```

Comment:
Fields should not be used for storing context information, please provide them as arguments to the methods.